metrics-core/src/main/java/com/codahale/metrics/ChunkedAssociativeLongArray.java

```diff
@@ -1,95 +1,87 @@
 package com.codahale.metrics;
 
+import static java.lang.System.arraycopy;
 import static java.util.Arrays.binarySearch;
 
-import java.util.ArrayDeque;
-import java.util.Deque;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.ListIterator;
 
 class ChunkedAssociativeLongArray {
     private static final long[] EMPTY = new long[0];
     private static final int DEFAULT_CHUNK_SIZE = 512;
 
     private final int defaultChunkSize;
-    private Chunk activeChunk;
+    private final LinkedList<Chunk> chunks = new LinkedList<Chunk>();
```

Comment:
Agreed. I run the benchmak and it seems `ÀrrayList` doesn't provide any advantages. Probably because of removing from the middle of the list during trimming of chunks.